components/scream/src/share/tests/common_physics_functions_tests.cpp

```diff
@@ -0,0 +1,298 @@
+
+    using Spack = ekat::Pack<Scalar,SCREAM_SMALL_PACK_SIZE>;
+
+    int num_levs = 100; // Number of levels to use for tests.
+
+    static constexpr Scalar p0     = C::P0;
+    static constexpr Scalar Rd     = C::RD;
+    static constexpr Scalar inv_cp = C::INV_CP;
+    static constexpr Scalar tmelt  = C::Tmelt;
+    static constexpr Scalar ggr    = C::gravit;
+    static constexpr Scalar test_tol = C::macheps*1e2;
+
+    using view_1d = ekat::KokkosTypes<DefaultDevice>::view_1d<Scalar>;
+    using sview_1d = ekat::KokkosTypes<DefaultDevice>::view_1d<Spack>;
+    const Int num_pack = ekat::npack<Spack>(num_levs);
+    const Int num_pack_int = ekat::npack<Spack>(num_levs+1);
+    // Compute random values for tests
+    sview_1d temperature_packed("temperature",num_pack),
+             height_packed("height",num_pack),
+             qv_packed("qv",num_pack),
+             pressure_packed("pressure",num_pack),
+             pseudo_density_packed("pseudo_density",num_pack),
+             dz_for_testing_packed("dz_for_testing",num_pack);
+    view_1d surface_height("surface_height",1);
+    // Allocate memory for test outputs
+    view_1d exner("exner",num_levs),
+            theta("theta",num_levs),
+            T_mid_from_pot("T_mid_from_pot",num_levs),
+            T_virtual("T_virtual",num_levs),
+            T_mid_from_virt("T_mid_from_virt",num_levs),
+            dse("dse",num_levs),
+            dz("dz",num_levs),
+            z_int("z_int",num_levs+1);
+    sview_1d exner_packed("exner",num_pack),
+             theta_packed("theta",num_pack),
+             T_mid_from_pot_packed("T_mid_from_pot",num_pack),
+             T_virtual_packed("T_virtual",num_pack),
+             T_mid_from_virt_packed("T_mid_from_virt",num_pack),
+             dse_packed("dse",num_pack),
+             dz_packed("dz",num_pack),
+             z_int_packed("z_int",num_pack_int);
+
+    // Construct random input data
+    std::random_device rdev;
+    using rngAlg = std::mt19937_64;
+    const unsigned int catchRngSeed = Catch::rngSeed();
+    const unsigned int seed = catchRngSeed==0 ? rdev() : catchRngSeed;
+    rngAlg engine(seed);
+    using RPDF = std::uniform_real_distribution<Scalar>;
+    RPDF pdf_qv(1e-3,1e3),
+         pdf_dp(1.0,100.0),
+         pdf_pres(0.0,p0),
+         pdf_temp(200.0,400.0),
+         pdf_height(0.0,1e5),
+         pdf_surface(100.0,400.0);
+
+    ekat::genRandArray(temperature_packed,engine,pdf_temp);
+    ekat::genRandArray(height_packed,engine,pdf_height);
+    ekat::genRandArray(surface_height,engine,pdf_surface);
+    ekat::genRandArray(qv_packed,engine,pdf_qv);
+    ekat::genRandArray(pressure_packed,engine,pdf_pres);
+    ekat::genRandArray(pseudo_density_packed,engine,pdf_dp);
+
+    // Construct a simple set of `dz` values for testing the z_int function
+    auto dz_for_testing_host = Kokkos::create_mirror(dz_for_testing_packed);
+    for (int k = 0;k<num_levs;k++)
```

Comment:
Once you have the scalarized view, you could even do:
```
std::iota(dz_for_testing_host.data(),dz_for_testing_host.data()+num_levs,1); // gen sequence 1,2,...
std::reverse(dz_for_testing_host.data(),dz_for_testing_host.data()+num_levs); // revert sequence (b/c of atm lev ordering)
```
Here it's best to leave the manual for loop. I just thought I mentioned std library functions, in case you were not aware of those.

Reply:
So be "scalarizing" the view I should be able to replace `dz_for_testing_host(ipack)[ivec]` with `dz_for_testing_host[k]`?

Reply:
Yes.